task_processing/plugins/mesos/execution_framework.py

```diff
@@ -101,11 +101,22 @@ def __init__(
         }
 
         self.stopping = False
+
         task_kill_thread = threading.Thread(
             target=self.kill_tasks_stuck_in_staging, args=())
         task_kill_thread.daemon = True
         task_kill_thread.start()
 
+        # metadata is not removed immediately upon terminal event because mesos
```

Comment:
If we receive a terminal status update for a task, then we remove it from the internal state immediately. I am not sure if this review was created before I added the fix for duplicate terminal events. Currently, we ignore duplicate status updates. Do we still need this review?